loader/src/main/java/net/neoforged/fml/loading/mixin/FMLMixinLaunchPlugin.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.fml.loading.mixin;
+
+import com.mojang.logging.LogUtils;
+import cpw.mods.modlauncher.api.ITransformerActivity;
+import cpw.mods.modlauncher.api.NamedPath;
+import cpw.mods.modlauncher.serviceapi.ILaunchPluginService;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import net.neoforged.fml.loading.LoadingModList;
+import net.neoforged.fml.loading.moddiscovery.ModFileInfo;
+import net.neoforged.fml.loading.moddiscovery.ModFileParser;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.ClassNode;
+import org.slf4j.Logger;
+import org.spongepowered.asm.launch.Phases;
+import org.spongepowered.asm.mixin.FabricUtil;
+import org.spongepowered.asm.mixin.MixinEnvironment;
+import org.spongepowered.asm.mixin.Mixins;
+import org.spongepowered.asm.mixin.transformer.Config;
+
+public class FMLMixinLaunchPlugin implements ILaunchPluginService {
+    public static final String NAME = "fml-mixin";
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    private MixinFacade facade;
+
+    public FMLMixinLaunchPlugin() {
+        System.setProperty("mixin.service", FMLMixinService.class.getName());
+        System.setProperty("mixin.bootstrapService", FMLMixinServiceBootstrap.class.getName());
+    }
+
+    public synchronized void setup(List<String> extraMixinConfigs) {
+        if (this.facade == null) {
+            this.facade = new MixinFacade(this);
+        }
+
+        Map<String, String> configModIds = new HashMap<>();
+        var modList = LoadingModList.get();
+
+        extraMixinConfigs.forEach(Mixins::addConfiguration);
+
+        modList.getModFiles().stream()
+                .map(ModFileInfo::getFile)
+                .forEach(file -> {
+                    final String modId = file.getModInfos().getFirst().getModId();
+                    for (ModFileParser.MixinConfig potential : file.getMixinConfigs()) {
+                        var existingModId = configModIds.putIfAbsent(potential.config(), modId);
+                        if (existingModId != null && !existingModId.equals(modId)) {
+                            LOGGER.error("Mixin config {} is registered by multiple mods: {} and {}", potential.config(), existingModId, modId);
+                            throw new IllegalStateException("Mixin config " + potential.config() + " is registered by multiple mods: " + existingModId + " and " + modId);
```

Comment:
What exactly was the old behaviour here? I think throwing is most sensible (if we can't _fix_ this somehow, that is...) but it might be good for this to be non-breaking, so that could happen later.

Reply:
And another point -- since this is an issue we _might_ actually expect to pop up in prod (unlike, say, the exceptions thrown for "the FMLMixinLaunchPlugin just doesn't exist" or the like), should use the ModLoadingIssue setup instead?